aten/src/ATen/native/RangeFactories.cpp

```diff
@@ -0,0 +1,35 @@
+#include <ATen/NativeFunctions.h>
+#include <ATen/Parallel.h>
+#include <ATen/Dispatch.h>
+
+namespace at { namespace native {
+
+
+Tensor& linspace_cpu_out(Tensor& result, Scalar start, Scalar end, int64_t steps) {
+  AT_CHECK(steps >= 0, "number of steps must be non-negative");
+
+  if (result.numel() != steps) {
+    result.resize_({steps});
+  }
+  Tensor r = result.is_contiguous() ? result : result.contiguous();
+
+  AT_DISPATCH_FLOATING_TYPES(r.type(), "linspace", [&]() {
+    scalar_t scalar_start = start.to<scalar_t>();
+    scalar_t scalar_end = end.to<scalar_t>();
+    scalar_t *data_ptr = r.data<scalar_t>();
+    scalar_t step = (scalar_end - scalar_start) / static_cast<scalar_t>(steps - 1);
+    at::parallel_for(0, steps, internal::GRAIN_SIZE, [&](int64_t p_begin, int64_t p_end) {
```

Comment:
It could be better to write exactly what you're capturing here to prevent someone in the future from adding something that can get accidentally caught and will slow things down.